packages/grid/src/Grid.tsx

```diff
@@ -1907,6 +1907,16 @@ class Grid extends PureComponent<GridProps, GridState> {
       metrics.rowHeight
     );
 
+    // Check if we're at the edge of the grid and attempting to scroll
```

Comment:
@mofojed this works for me with all cases I tested. I'm not very familiar with how `lastTop` and `lastLeft` are actually implemented or if there are nuances here we are missing. Any concerns on your part?

Reply:
There is one nuance that's missing. To give a little bit of context:
- The way we handle scrolling is by keeping track of the _cell_ index that is in the top left, and then keeping track of the pixel _offset_ within that cell. We do that so we can support very large grids and grids with variable width/height cells (since the width of a column may change after we've fetched data for it, we resize to match the data).
- `left`/`top` are the index of the cell in the top left
- `leftOffset`/`topOffset` is the pixel offset _within_ that cell that's in the top left
- Current scroll position is the position of the top/left cell + the offset

Here we're just checking `left`, but you're not checking `leftOffset` or `topOffset`. You should check if `(left > 0 || leftOffset > 0) && deltaX < 0` to check if you can scroll left (same with the top).

Actually just to be safe - I'd just track the initial values for all those variables `left`, `top`, `leftOffset`, and `topOffset`, and then just at the end of this method before `this.setViewState` check if they haven't changed, and if they haven't, don't bother setting them or eating the event. That seems like the safest.

Reply:
I think @mofojed suggestion here was to remove this section and just have the check at the end that you added below. I'll let him confirm, but I tested again using only the code at the end, and that seems to work as expected. I don't think we need this code at the beginning anymore.

Reply:
Ahh I might have misinterpreted, I thought just to be safe meant that the check at the bottom would be a failsafe, not the only check. Will test by removing initial code and repush accordingly.